site/content/en/docs/Reference/API/Kustomization File/configMapGenerator.md

```diff
@@ -6,3 +6,81 @@ weight: 6
 description: >
     Generate ConfigMap resources.
 ---
+
+## configMapGenerator
+
+ConfigMap objects can be generated by adding a `configMapGenerator` entry to the `kustomization.yaml` file.
+
+See [Generating ConfigMaps Tasks] for examples of how to use `configMapGenerator`.
+
+---
+
+* **apiVersion**: kustomize.config.k8s.io/v1beta1
+* **kind**: Kustomization
+* **configMapGenerator** ([][ConfigMapArgs](#configmapargs))
+
+    List of ConfigMaps to generate.
+
+
+### ConfigMapArgs
+
+ConfigMapArgs contains the metadata of the generated ConfigMap.
+
+* **name** (string)
+
+    Name of the generated ConfigMap. A hash suffix will be added to the Name by default.
+
+* **namespace** (string)
```

Comment:
Probably good to point out optional fields. @koba1t suggested this: https://github.com/kubernetes-sigs/kustomize/pull/5247#issuecomment-1690444816

Reply:
I added an "optional" to the applicable fields. It looks like upstream [Kubernetes docs](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers) typically notes which fields are required and omits the optional label. I don't feel strongly about either style so long as we are consistent.